endpoints/openrtb2/amp_auction.go

```diff
@@ -382,16 +380,15 @@ func (deps *endpointDeps) overrideWithParams(httpRequest *http.Request, req *ope
 		req.Imp[0].TagID = slot
 	}
 
-	privacyPolicies := privacy.Policies{
-		GDPR: gdpr.Policy{
-			Consent: httpRequest.URL.Query().Get("gdpr_consent"),
-		},
-		CCPA: ccpa.Policy{
-			Value: httpRequest.URL.Query().Get("us_privacy"),
-		},
-	}
-	if err := privacyPolicies.Write(req); err != nil {
-		return err
+	consent := readConsent(httpRequest.URL)
+	if consent != "" {
+		if policies, ok := privacy.ReadPoliciesFromConsent(consent); ok {
+			if err := policies.Write(req); err != nil {
+				return err
+			}
+		} else {
+			// TODO: Send warning to caller.
```

Comment:
Not sure where this warning should go. Asked for clarification in linked issue.

Reply:
> We could maybe make privacy.ReadPoliciesFromConsent return an error instead of a bool so we can channel its message into our potential warning system:

I'm afraid that won't work in this case. `ReadPoliciesFromConsent` attempts to parse the consent string as both CCPA and TCF, since AMP doesn't tell us which one to expect. The detailed error messages are of little use if we aren't sure which format we're supposed to be targeting. The best I think we can do here is to fill in the TODO with a warning that the format was unrecognized. Other parts of code either currently use or may in the future use the message of the error.